### PR TITLE
[SPARK-30234][SQL][FOLLOWUP] Rename `spark.sql.legacy.addDirectory.recursive.enabled` to `spark.sql.legacy.addSingleFileInAddFile`

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?
Rename `spark.sql.legacy.addDirectory.recursive.enabled` to `spark.sql.legacy.addSingleFileInAddFile`

### Why are the changes needed?
To follow the naming convention

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Existing UTs.

### CODE REVIEW

**Comment by iRakson** [comment]
@cloud-fan @Ngone51 @maropu

**Comment by dongjoon-hyun** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -2210,13 +2210,13 @@ object SQLConf {
...
+  val LEGACY_ADD_SINGLE_FILE_IN_ADD_FILE =
+    buildConf("spark.sql.legacy.addSingleFileInAddFile")
       .internal()
-      .doc("When true, users can add directory by passing path of a directory to ADD FILE " +
-        "command of SQL. If false, then only a single file can be added.")
+      .doc("When true, only a single file can be added using ADD FILE. If false, then users " +
+        "can add directory by passing directory path to ADD FILE.")
       .booleanConf
-      .createWithDefault(true)
+      .createWithDefault(false)
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: ? Why do you try to change the default value in `Rename ...` PR title?

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Yea, I think this comes from the comment: https://github.com/apache/spark/pull/26863#discussion_r378715586

Comment (dongjoon-hyun) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Thanks, @maropu .

**Comment by HyukjinKwon** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Review by Ngone51 — COMMENTED** [review]
I'm actually OK with this change as it fix the inconsistent of the config.

But I still doubt whether we should treat it as legacy, here. To me, legacy just sounds like the previous behavior is not good, especially there's some problem with it. And, in practical, I believe Spark would aways recommend user to use the new behavior over the legacy behavior.

But, in this case, does Spark has a strong prefer to recommend user to use add directory when using `ADD FILE`? I think it depends on use case, right?

So, personally, I still think the config should be a new feature flag rather than legacy. And we can make it turn on by default to be friendly to user.

**Comment by HyukjinKwon** [comment]
>  To me, legacy just sounds like the previous behavior is not good, especially there's some problem with it.

I doesn't necessarily have to have a problem with it. We have a legacy configuration based on case-by-case, for example, when the previous behaviour has been widely used, newer behaviour supersedes the old behaviour, etc. Note that bugs have to be problem without legacy config as a general guideline!

`spark.sql.legacy.addSingleFileInAddFile` supersedes old behaviour, and we will remove this configuration away in the future because disabling `spark.sql.legacy.addSingleFileInAddFile` will support **both a directory and single file**.

Also, I don't actually completely buy this argument https://github.com/apache/spark/pull/26863#discussion_r378715586 since we already have different default values for legacy configurations, e.g., `spark.sql.legacy.replaceDatabricksSparkAvro.enabled` vs `spark.sql.legacy.setopsPrecedence.enabled`. I only buy the argument about the namespace. Doesn't quite matter if it's called directory or file.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by HyukjinKwon** [comment]
> But, in this case, does Spark has a strong prefer to recommend user to use add directory when using ADD FILE? I think it depends on use case, right?

Spark will have support a directory in `ADD FILE` as a Hive compatibility concern as discussed in https://github.com/apache/spark/pull/26863.

**Pull request closed without merging**